src/control/server/storage/bdev/backend.go

```diff
@@ -127,16 +140,16 @@ func hugePageWalkFunc(hugePageDir, prefix, tgtUID string, remove removeFn, count
 				return nil
 			}
 			return filepath.SkipDir // skip subdirectories
-		case !strings.HasPrefix(info.Name(), prefix):
-			return nil // skip files without prefix
 		}
 
-		stat, ok := info.Sys().(*syscall.Stat_t)
-		if !ok || stat == nil {
-			return errors.New("stat missing for file")
+		matches := re.FindStringSubmatch(info.Name())
+		if len(matches) != 2 {
+			return nil // skip files not matching expected pattern
 		}
-		if strconv.Itoa(int(stat.Uid)) != tgtUID {
-			return nil // skip not owned by target user
+		// PID string will be the first submatch at index 1 of the match results.
+
+		if isActive, err := isPIDActive(matches[1], stat); err != nil || isActive {
+			return err // skip files created by an existing process (isActive == true)
```

Comment:
My only concern with this is that an error here will cause server startup to fail. I'm fine with having `isPIDActive()` return an error so that the caller can decide, but my inclination would be that an error in checking for an active pid should just cause the file to be skipped. Otherwise, we may find that we've introduced a situation where the admin can't start the server and the error being dumped to the journal is kind of obscure.

Reply:
yes I have the same concern, an error gets logged but it doesn't prevent server start-up: https://github.com/daos-stack/daos/pull/9625/files#diff-d4337e9b4e14476c4e9211f64d51adcdfc20fe8f35ce486e450063d637ef1d7cR406